packages/ui/src/components/Charts/index.tsx

```diff
@@ -0,0 +1,285 @@
+"use client";
+
+import * as React from "react";
+import * as ChartPrimitive from "recharts";
+import { getTwColor } from "../../utils/getTwColor";
+
+import type { ChartConfig } from "./types";
+import { ChartContext, useChart } from "./hooks/useChart";
+import { getPayloadConfigFromPayload } from "./helpers/getPayloadConfigFromPayload";
+import { cn } from "../../utils/cn";
+import { THEMES } from "./constants";
+import { Text } from "../Text";
+
+const ChartContainer = React.forwardRef<
+  HTMLDivElement,
+  React.ComponentProps<"div"> & {
+    config: ChartConfig;
+    children: React.ComponentProps<typeof ChartPrimitive.ResponsiveContainer>["children"];
+  }
+>(({ id, className, children, config, ...props }, ref) => {
+  const uniqueId = React.useId();
+  const chartId = `chart-${id || uniqueId.replace(/:/g, "")}`;
+
+  return (
+    <ChartContext.Provider value={{ config }}>
+      <div
+        data-chart={chartId}
+        ref={ref}
+        className={cn(
+          "flex aspect-video justify-center text-xs [&_recharts-cartesian-axis]:font-[inherit] [&_.recharts-cartesian-axis-tick_text]:fill-slate-600/60 [&_.recharts-cartesian-grid_line[stroke='#ccc']]:stroke-slate-300/50 [&_.recharts-curve.recharts-tooltip-cursor]:stroke-border [&_.recharts-dot[stroke='#fff']]:stroke-transparent [&_.recharts-layer]:outline-none [&_.recharts-polar-grid_[stroke='#ccc']]:stroke-border [&_.recharts-radial-bar-background-sector]:fill-muted [&_.recharts-rectangle.recharts-tooltip-cursor]:fill-muted [&_.recharts-reference-line_[stroke='#ccc']]:stroke-border [&_.recharts-sector[stroke='#fff']]:stroke-transparent [&_.recharts-sector]:outline-none [&_.recharts-surface]:outline-none",
+          className,
+        )}
+        {...props}
+      >
+        <ChartStyle id={chartId} config={config} />
+        <ChartPrimitive.ResponsiveContainer>{children}</ChartPrimitive.ResponsiveContainer>
+      </div>
+    </ChartContext.Provider>
+  );
+});
+
+ChartContainer.displayName = "Chart";
+
+const ChartStyle = ({ id, config }: { id: string; config: ChartConfig }) => {
+  const colorConfig = Object.entries(config).filter(([_, config]) => config.theme || config.color);
+
+  if (!colorConfig.length) {
+    return null;
+  }
+
+  return (
+    <style
+      // biome-ignore lint/security/noDangerouslySetInnerHtml: It's on purpose
+      dangerouslySetInnerHTML={{
+        __html: Object.entries(THEMES)
+          .map(
+            ([theme, prefix]) => `
+${prefix} [data-chart=${id}] {
+${colorConfig
+  .map(([key, itemConfig]) => {
+    const color = itemConfig.theme?.[theme as keyof typeof itemConfig.theme] || itemConfig.color;
+    return color ? `  --color-${key}: ${getTwColor(color)};` : null;
+  })
+  .filter(Boolean)
+  .join("\n")}
+}
+`,
+          )
+          .join("\n"),
+      }}
+    />
+  );
+};
+
+const ChartTooltipContent = React.forwardRef<
```

Comment:
In the tooltip I would add the corresponding color of each case.

Reply:
Also a small space is needed
![Uploading Screenshot 2024-09-09 at 17.24.51.png…]()
 between values and label in the tooltip